ui/apps/dashboard/e2e/member clusters/test-utils.ts

```diff
@@ -0,0 +1,228 @@
+import * as k8s from '@kubernetes/client-node';
+import { parse } from 'yaml';
+import _ from 'lodash';
+
+// Set webServer.url and use.baseURL with the location of the WebServer
+const HOST = process.env.HOST || 'localhost';
+const PORT = process.env.PORT || 5173;
+const baseURL = `http://${HOST}:${PORT}`;
+const basePath = '/multicloud-resource-manage';
+const token = process.env.KARMADA_TOKEN || '';
+
+// Karmada API server configuration - can be overridden by environment variables
+const KARMADA_HOST = process.env.KARMADA_HOST || 'localhost';
+const KARMADA_PORT = process.env.KARMADA_PORT || '5443'; // Standard Karmada API server port
+const KARMADA_API_SERVER = `https://${KARMADA_HOST}:${KARMADA_PORT}`;
+
+/**
+ * Creates a configured Kubernetes API client for Karmada
+ * @returns Kubernetes CustomObjectsApi client
+ */
+function createKarmadaApiClient(): k8s.CustomObjectsApi {
+    const kc = new k8s.KubeConfig();
+
+    // Try to use existing kubeconfig first (for CI)
+    if (process.env.KUBECONFIG) {
+        try {
+            kc.loadFromFile(process.env.KUBECONFIG);
+            // Set context to karmada-apiserver if available
+            const contexts = kc.getContexts();
+            const karmadaContext = contexts.find(c => c.name === 'karmada-apiserver');
+            if (karmadaContext) {
+                kc.setCurrentContext('karmada-apiserver');
+            }
+            return kc.makeApiClient(k8s.CustomObjectsApi);
+        } catch (error) {
+            console.error('Failed to load kubeconfig:', error);
+        }
+    }
+
+    // Fallback to custom config for local development
+    const kubeConfigYaml = `
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    insecure-skip-tls-verify: true
+    server: ${KARMADA_API_SERVER}
+  name: karmada-apiserver
+contexts:
+- context:
+    cluster: karmada-apiserver
+    user: karmada-dashboard
+  name: default
+current-context: default
+users:
+- name: karmada-dashboard
+  user:
+    token: ${token}
+`;
+
+    kc.loadFromString(kubeConfigYaml);
+    return kc.makeApiClient(k8s.CustomObjectsApi);
+}
+
+/**
+ * Setup dashboard authentication and navigate to dashboard page
+ */
+export async function setupDashboardAuthentication(page: Page) {
+    await page.goto(`${baseURL}/login`, { waitUntil: 'networkidle' });
+    await page.evaluate((t) => localStorage.setItem('token', t), token);
+    await page.goto(`${baseURL}${basePath}`, { waitUntil: 'networkidle' });
+    await page.evaluate((t) => localStorage.setItem('token', t), token);
+    await page.reload({ waitUntil: 'networkidle' });
+    await page.waitForSelector('text=Overview', { timeout: 30000 });
+}
+
+/**
+ * Generate test Member Cluster kubeconfig YAML with timestamp
+ * Note: The Member Cluster creation page expects kubeconfig format, not Cluster resource format
+ */
+export function generateTestMemberClusterYaml() {
+    const timestamp = Date.now();
+    return `apiVersion: v1
```

Comment:
A bit curious, can we join the cluster without needing the cluster's certificate and key?

Reply:
@XiShanYongYe-Chang here in the kubeconfig, specify `insecure-skip-tls-verify: true` in `cluster` field